.github/workflows/docker.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches: [main]
```

Comment:
why main? or tag. I would suggest to use 

```
on:
  release:
    types: [published]

```

Reply:
When you create a release, you will be pushing it with a tag. Running the release on a pushed tag will allow us to specify the tag number as the docker image tag. This enables us to easily use the tagged image in our cluster.

We want the workflow to trigger on push to `main` branch so that we can test this image on our cluster.

Reply:
Okay, that sounds fair enough